app/js/lib/foxbox/event-dispatcher.js

```diff
@@ -206,42 +196,25 @@ export default class EventDispatcher {
       return;
     }
 
-    handlers.forEach(function(handler) {
+    handlers.forEach((handler) => {
       try {
-        handler(parameters);
+        handler.call(this, parameters);
       } catch (error) {
         console.error(error);
       }
     });
   }
-}
 
-/**
- * Mixes dispatcher methods into target object.
- * @param {Object} target Object to mix dispatcher methods into.
- * @param {Array.<string>} allowedEvents Optional list of the allowed event
- * names that can be emitted and listened for.
- * @returns {Object} Target object with added dispatcher methods.
- */
-EventDispatcher.mixin = function(target, allowedEvents) {
-  if (!target || typeof target !== 'object') {
-    throw new Error('Object to mix into should be valid object!');
-  }
+  /**
+   * Checks if there are any listeners that listen for the specified event.
+   *
+   * @param {string} eventName Name of the event to check listeners for.
+   * @returns {boolean}
+   */
+  hasListeners(eventName) {
```

Comment:
I have mixed feelings about this method - but it simplifies a lot my next PR, so I've added this one. If in the future I find better approach to know that I can shutdown the bus (or something that depends on the bus) if there is no listeners anymore, I would probably try to get rid of this method.